CRM/Utils/Array.php

```diff
@@ -1058,16 +1058,20 @@ public static function formatArrayKeys(&$array) {
     if (!is_array($array)) {
       return;
     }
-    $keys = array_keys($array, 1);
-    if (count($keys) > 1 ||
-      (count($keys) == 1 &&
-        (current($keys) > 1 ||
-          is_string(current($keys)) ||
-          (current($keys) == 1 && $array[1] == 1) // handle (0 => 4), (1 => 1)
-        )
-      )
-    ) {
-      $array = $keys;
+
+    $hasSequentialKeysStartingWithZero = array_keys($array) === range(0, count($array) - 1);
+    $hasAllValuesAsNumberOne = array_unique(array_values($array)) === ['1'];
```

Comment:
=== is a problem here, if the result of array_unique(array_values($array)) was [1] then this will result in False since [1] !== ['1']

Reply:
I am trying to avoid using `==` at all times. Maybe that habit goes from my JS experience, but I prefer conversion and strict comparison in order to avoid something like `0 == FALSE`